### PR TITLE
Add serialization for TimestampedTransform.

### PR DESCRIPTION
**Pull request by wohe**

### CODE REVIEW

File: cartographer/transform/timestamped_transform_test.cc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+namespace {
+
+TEST(TimestampedTransformTest, ToProtoAndBack) {
+  const TimestampedTransform expected{
+      common::FromUniversal(12345678),
+      Rigid3d(Eigen::Vector3d(1., 2., 3.),
+              Eigen::Quaterniond(1., 2., 3., 4.).normalized())};
+  const TimestampedTransform actual = FromProto(ToProto(expected));
+  EXPECT_EQ(expected.time, actual.time);
+  EXPECT_THAT(actual.transform, IsNearly(expected.transform, 1e-6));
```
Comment (danielsievers) [inline cartographer/transform/timestamped_transform_test.cc]: Should be exactly the same right?

Comment (wohe) [reply cartographer/transform/timestamped_transform_test.cc]: If we assume IEEE-754, I think so. But `IsNearly` is readily available while `operator==` is not, and it gives us good enough coverage.

**Comment by wohe** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user wohe. Merge queue now has a length of 1.

**Pull request merged**